sourcegraph/sourcegraph.proto

```diff
@@ -102,6 +102,10 @@ message Repo {
 	// URL. E.g., "github.com/user/repo".
 	string uri = 1 [(gogoproto.customname) = "URI"];
 
+	// Origin is populated for repos fetched via federation or
+	// discovery. It is the hostname of the host that owns the repo.
+	string origin = 21;
```

Comment:
A non-empty origin means the repo was fetched via discovery and federation.

You might have a repo with URI golang.org/x/tools whose clone URL is https://googlesource.com/golang/tools (just making that up, I'm not at my computer). Since golang.org doesn't run Sourcegraph, the data for the tools repo would be fetched from Sourcegraph.com as a special case (discovery using Go vanity import path lookup procedure). In that case I think origin would be Sourcegraph.com, URI would be golang.org/x/tools, and clone URL would be https://googlesource.com/golang/tools.

Why have an Origin field?

- It tells the server whether to make the repo available for discovery. You can browse to https://internal-team1.sourcegraph.mycompany.local/golang.org/x/tools and it will fetch the data from Sourcegraph.com via federation. But if another team in the same company tries to access https://internal-team55.sourcegraph.mycompany.local/internal-team1.sourcegraph.mycompany.local/golang.org/x/tools on their different Sourcegraph instance, it should know that that is not the repo's canonical source and it should not make that available for discovery. 
- It lets the UI show a little tag indicating that the repo was fetched remotely via federation (like "django/django via github.com").

I am not sure if this is the best way to do it. What do you think?

> On Jun 20, 2015, at 12:51, Dmitri Shuralyov <notifications@github.com> wrote:
> 
> In sourcegraph/sourcegraph.proto:
> 
> > @@ -102,6 +102,10 @@ message Repo {
> >  	// URL. E.g., "github.com/user/repo".
> >  	string uri = 1 [(gogoproto.customname) = "URI"];
> >  
> > +	// Origin is populated for repos fetched via federation or
> > +	// discovery. It is the hostname of the host that owns the repo.
> > +	string origin = 21;
> This origin field, what is it's relationship to the clone_url in ReposCreateOp? Do they have different names because of a subtle different meaning, or simply because they were created at different times?
> 
> If mirror is true, does that mean origin is effectively the clone_url of the mirror's canonical location?
> 
> —
> Reply to this email directly or view it on GitHub.
>